src/smallestPrimeFactor.js

```diff
@@ -11,13 +11,45 @@
  * @param {number} num - A number
  * @returns {number} The smallest prime factor of num
  */
+function isPrime(num) {
+  if(num===1){
+    return false;
+  }
+  if(num===2) {
+    return true;
+  }
+  for(var i=2;i<num;i++){
+    if(num%i===0){
+      return false;
+    }
+
+  }
+  return true;
+}
 function smallestPrimeFactor(num) {
-  // This is your job. :)
+  if(isPrime(num)){
+    return num;
+  }
+  for(let i = 1; i<num;i++ ) {
```

Comment:
Trace what this does for a number like `7 * 19 * 23 ===  3059`. It'll iterate up to `7`(the smallest factor) and then call `isPrime(7)`. What does `isPrime(7)` do?